### PR TITLE
Alfoa/issue1199

### LINKED ISSUE #1199: [TASK] ROM Clustering improvement
--------
Issue Description
--------

Improvements that are needed in ROM clustering.
In addition to #913 

 - [ ] ROM selection:  User should be able to specify the method for selecting on which data point in each cluster, this data point should be a sufficient representation of the clustered data.

The current evaluation after clustering is taking the first data point in the cluster as the cluster typical point. This might lead to all data points in the same cluster represented by an unnormal point if we believe there is a wide variation in the cluster itself. 

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [ ] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

Closes #1199

##### What are the significant changes in functionality due to this change request?
activated the possibility to randomly pick a ROM withing a cluster for the evaluation stage of the time series.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: doc/user_manual/rom.tex [hunk 1]
```diff
@@ -85,6 +85,16 @@
...
         \end{itemize}
+     \item \xmlNode{evaluationClusterChoice}, \xmlDesc{string, optional field}, one of \xmlString{first},
+        \xmlString{random}, or \xmlString{centroid}, determines, if \xmlAttr{grouping}$=cluster$, which 
+        strategy needs to be followed for the evaluation stage. If ``first'', the 
+        first ROM (representative segmented ROM),in each cluster, is considered to
+         be representative of the full space in the cluster (i.e. the evaluation is always performed
+         interrogating the first ROM in each cluster); If ``random'', a random ROM, in each cluster, 
+         is choosen when an evaluation is requested; If ``centroid'', a ROM ``trained" on the centroids 
+         information of each cluster is used for the evaluation (\nb ``centroid'' option is not 
+         available yet).
```
Comment (JiaZhou-PU) [inline doc/user_manual/rom.tex:96]: why adding the centroid option here? If it is not available...

File: tests/framework/ROM/TimeSeries/ARMA/clustered_choice_rand.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      </options>
+    </HistorySet>
+    <HistorySet name="samples">
+      <Input>scaling</Input>
+      <Output>Signal, Time</Output>
+      <options>
+        <pivotParameter>Time</pivotParameter>
+      </options>
+    </HistorySet>
+    <DataSet name="romMeta"/>
```
Comment (JiaZhou-PU) [inline tests/framework/ROM/TimeSeries/ARMA/clustered_choice_rand.xml:72]: Should remove this, or add a print in this data set.

Comment (alfoa) [reply tests/framework/ROM/TimeSeries/ARMA/clustered_choice_rand.xml:72]: removed

File: doc/user_manual/rom.tex (hunk 1 above)
Comment (alfoa) [reply doc/user_manual/rom.tex:96]: you are right...I remove it.

**Comment by alfoa** [comment]
@JiaZhou-PU addressed your comments

**Comment by JiaZhou-PU** [comment]
PR approved. Checklist passed.

**Pull request merged**